src/Microsoft.AspNet.JsonPatch/Adapters/ObjectAdapter.cs

```diff
@@ -133,9 +134,9 @@ private void Add(string path, object value, T objectToApplyTo, Operation<T> oper
                 // what if it's an array but there's no position??
                 if (IsNonStringArray(patchProperty))
                 {
-                    // now, get the generic type of the enumerable
+                    // now, get the generic type of the enumerable from the runtime type of patchProperty value.
                     var genericTypeOfArray = PropertyHelpers.GetEnumerableType(
```

Comment:
Should we be looking for an `IList<>` type instead? That's really the contract that we need if we're adding/inserting.